src/message_replies/givePoint.js

```diff
@@ -2,65 +2,92 @@ import { getUserIdFromMention } from '../utils/getUserIdFromMention.js';
-  const mentionId = getUserIdFromMention(command);
+const ARRAY_OF_POINTS = ['1️⃣', '2️⃣', '3️⃣', '4️⃣', '5️⃣', '6️⃣', '7️⃣', '8️⃣', '9️⃣'];
+
+async function givePoint(commandArr, interaction) {
   const caller = interaction.author.id;
 
-  if (!mentionId) {
-    return interaction.reply(
-      `Sorry <@!${caller}>, can't find ${command}.\n(╯°□°）╯︵ ┻━┻`
+  // check if ids are valid or not and returns an array of objects
+  // the objects have shape {isPossible: boolean, ID: string}
+  let mentionIDs = commandArr.map((command) => getUserIdFromMention(command));
+
+  // filter those that are repeated
+  mentionIDs = mentionIDs.reduce((acc, IDObj) => {
+    // removes those who's id is alread in the array and grab the array length
+    const filteredLength = acc.filter(({ id }) => id !== IDObj.id).length;
+
+    if (acc.length === filteredLength) {
+      acc.push(IDObj);
+      return acc;
+    }
+    return acc;
+  }, []);
+
+  if (mentionIDs.length > 9) {
+    mentionIDs = mentionIDs.slice(0, 9);
+
+    await interaction.reply(
+      `Sorry <@!${caller}>, You can only give points to 9 people per message.
+Points will be given ONLY for the first 9 people mentioned`
     );
   }
 
-  const { validUser, username } = await isUserValid(interaction, mentionId);
+  // maintain only those that are valid
+  const validIDs = mentionIDs.filter((mentionID) => mentionID.isPossibleID);
 
-  // <@123456789012345678> will be an id but not a valid one,
-  // therefore will need to be checked, and will need this message;
-  if (!validUser) {
-    interaction.reply(
-      // eslint-disable-next-line no-useless-escape
-      `Sorry <@!${caller}>, idk who ${command} is. ¯\\_(ツ)_/¯`
+  // checks ID's against discord and keeps only those that are valid in the discord API
+  let discordVerified = [];
+  for (const idObj of validIDs) {
+    const { validUser, username: name } = await isUserValid(
+      interaction,
+      idObj.id
     );
-  } else if (caller === mentionId) {
+    if (validUser) {
+      discordVerified.push({ ...idObj, name });
+    }
+  }
+
+  // finding if is someone has the same id than the caller.
+  const hasCallerMention = discordVerified.some(({ id }) => caller === id);
+
+  if (hasCallerMention) {
+    // filters the caller out
+    discordVerified = discordVerified.filter(({ id }) => id !== caller);
     interaction.reply(`Lmao <@!${caller}>, you can't give yourself a point.`);
-  } else {
-    // try to add the user to the DB, if they are already there
-    // db function will reject this
+  }
 
-    await BuddiesModel.addUserToPoints(mentionId);
+  // keeps track the total points given and those who weren't
+  let givenPoints = 0;
+  const notGivenPoints = [];
 
-    const canAddPoint = await BuddiesModel.testDates(mentionId, interaction);
+  //  check agaist the DB if the date is valid (if last points was more than 1min ago)
+  for (const { id, name } of discordVerified) {
+    await BuddiesModel.addUserToPoints(id);
```

Comment:
We dont need to add a user here, do we? Won't we attempt to add a user to the db inside of `giveUserAPoint` on line 71?

This feels like it could slow us down a bit, it has to make extra DB queries unnecessarily.

Reply:
indeed. I also thought the same, but kept there because it was there before and I wasn't sure why. good catch